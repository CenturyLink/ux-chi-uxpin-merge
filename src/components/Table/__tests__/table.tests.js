import React from 'react';
import * as enzyme from 'enzyme';
import toJson from 'enzyme-to-json';
import Table from '../Table';

test('renders table with width stretched correctly', () => {
  const table = (
    <Table
      width="stretched"
      data={{
        header: ['band', 'singer', 'rhythm guitar', 'lead guitar', 'bass', 'drums', 'keyboard'],
        body: [
          {
            band: 'metallica',
            singer: 'james hatfield',
            'rhythm guitar': 'james hatfield',
            'lead guitar': 'kirk hammet',
            bass: 'robert trujillo',
            drums: 'lars ulrich',
          },
          {
            band: 'slayer',
            singer: 'tom araya',
            'rhythm guitar': 'kerry king',
            'lead guitar': 'gary holt',
            bass: 'tom araya',
            drums: 'paul bostaph',
          },
          {
            band: 'black sabbath',
            singer: 'ozzy osbourne',
            'lead guitar': 'tommy iommi',
            bass: 'geezer butler',
            drums: 'bill ward',
          },
          {
            band: 'queen',
            singer: 'freddy mercury',
            'lead guitar': 'brian may',
            bass: 'john deacon',
            drums: 'roger taylor',
            keyboard: 'freddy mercury',
          },
          {
            band: 'led zeppelin',
            singer: 'robert plant',
            'lead guitar': 'jimmy page',
            bass: 'john paul johns',
            drums: 'bonzo bonham',
            keyboard: 'john paul johns',
          },
          {
            band: 'deep purple',
            singer: 'ian gillan',
            'lead guitar': 'ritchie blackmore',
            bass: 'roger glover',
            drums: 'ian paice',
            keyboard: 'jon lord',
          },
        ],
      }}
    />
  );

  expect(toJson(enzyme.shallow(table))).toMatchSnapshot('enzyme.shallow');
});

test('renders table with width auto correctly', () => {
  const table = (
    <Table
      width="auto"
      data={{
        header: ['band', 'singer', 'rhythm guitar', 'lead guitar', 'bass', 'drums', 'keyboard'],
        body: [
          {
            band: 'metallica',
            singer: 'james hatfield',
            'rhythm guitar': 'james hatfield',
            'lead guitar': 'kirk hammet',
            bass: 'robert trujillo',
            drums: 'lars ulrich',
          },
          {
            band: 'slayer',
            singer: 'tom araya',
            'rhythm guitar': 'kerry king',
            'lead guitar': 'gary holt',
            bass: 'tom araya',
            drums: 'paul bostaph',
          },
          {
            band: 'black sabbath',
            singer: 'ozzy osbourne',
            'lead guitar': 'tommy iommi',
            bass: 'geezer butler',
            drums: 'bill ward',
          },
          {
            band: 'queen',
            singer: 'freddy mercury',
            'lead guitar': 'brian may',
            bass: 'john deacon',
            drums: 'roger taylor',
            keyboard: 'freddy mercury',
          },
          {
            band: 'led zeppelin',
            singer: 'robert plant',
            'lead guitar': 'jimmy page',
            bass: 'john paul johns',
            drums: 'bonzo bonham',
            keyboard: 'john paul johns',
          },
          {
            band: 'deep purple',
            singer: 'ian gillan',
            'lead guitar': 'ritchie blackmore',
            bass: 'roger glover',
            drums: 'ian paice',
            keyboard: 'jon lord',
          },
        ],
      }}
    />
  );

  expect(toJson(enzyme.shallow(table))).toMatchSnapshot('enzyme.shallow');
});

test('renders table with few headers correctly', () => {
  const table = (
    <Table
      width="auto"
      data={{
        header: ['band', 'singer', 'rhythm guitar'],
        body: [
          {
            band: 'metallica',
            singer: 'james hatfield',
            'rhythm guitar': 'james hatfield',
            'lead guitar': 'kirk hammet',
            bass: 'robert trujillo',
            drums: 'lars ulrich',
          },
          {
            band: 'slayer',
            singer: 'tom araya',
            'rhythm guitar': 'kerry king',
            'lead guitar': 'gary holt',
            bass: 'tom araya',
            drums: 'paul bostaph',
          },
          {
            band: 'black sabbath',
            singer: 'ozzy osbourne',
            'lead guitar': 'tommy iommi',
            bass: 'geezer butler',
            drums: 'bill ward',
          },
          {
            band: 'queen',
            singer: 'freddy mercury',
            'lead guitar': 'brian may',
            bass: 'john deacon',
            drums: 'roger taylor',
            keyboard: 'freddy mercury',
          },
          {
            band: 'led zeppelin',
            singer: 'robert plant',
            'lead guitar': 'jimmy page',
            bass: 'john paul johns',
            drums: 'bonzo bonham',
            keyboard: 'john paul johns',
          },
          {
            band: 'deep purple',
            singer: 'ian gillan',
            'lead guitar': 'ritchie blackmore',
            bass: 'roger glover',
            drums: 'ian paice',
            keyboard: 'jon lord',
          },
        ],
      }}
    />
  );

  expect(toJson(enzyme.shallow(table))).toMatchSnapshot('enzyme.shallow');
});
