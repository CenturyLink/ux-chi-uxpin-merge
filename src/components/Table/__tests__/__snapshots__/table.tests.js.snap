// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders table with few headers correctly: enzyme.shallow 1`] = `
<TableStyles
  data={
    Object {
      "body": Array [
        Object {
          "band": "metallica",
          "bass": "robert trujillo",
          "drums": "lars ulrich",
          "lead guitar": "kirk hammet",
          "rhythm guitar": "james hatfield",
          "singer": "james hatfield",
        },
        Object {
          "band": "slayer",
          "bass": "tom araya",
          "drums": "paul bostaph",
          "lead guitar": "gary holt",
          "rhythm guitar": "kerry king",
          "singer": "tom araya",
        },
        Object {
          "band": "black sabbath",
          "bass": "geezer butler",
          "drums": "bill ward",
          "lead guitar": "tommy iommi",
          "singer": "ozzy osbourne",
        },
        Object {
          "band": "queen",
          "bass": "john deacon",
          "drums": "roger taylor",
          "keyboard": "freddy mercury",
          "lead guitar": "brian may",
          "singer": "freddy mercury",
        },
        Object {
          "band": "led zeppelin",
          "bass": "john paul johns",
          "drums": "bonzo bonham",
          "keyboard": "john paul johns",
          "lead guitar": "jimmy page",
          "singer": "robert plant",
        },
        Object {
          "band": "deep purple",
          "bass": "roger glover",
          "drums": "ian paice",
          "keyboard": "jon lord",
          "lead guitar": "ritchie blackmore",
          "singer": "ian gillan",
        },
      ],
      "header": Array [
        "band",
        "singer",
        "rhythm guitar",
      ],
    }
  }
  width="auto"
>
  <thead>
    <HeaderRowStyles>
      <HeaderCellStyles
        key="0"
      >
        <HeaderCellWrapper>
          band
          <SortingStyles>
            <Button
              disabled={false}
              icon={
                <Icon
                  color="#f6f6f6"
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
            <Button
              disabled={false}
              icon={
                <Icon
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
          </SortingStyles>
        </HeaderCellWrapper>
      </HeaderCellStyles>
      <HeaderCellStyles
        key="1"
      >
        <HeaderCellWrapper>
          singer
          <SortingStyles>
            <Button
              disabled={false}
              icon={
                <Icon
                  color="#f6f6f6"
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
            <Button
              disabled={false}
              icon={
                <Icon
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
          </SortingStyles>
        </HeaderCellWrapper>
      </HeaderCellStyles>
      <HeaderCellStyles
        key="2"
      >
        <HeaderCellWrapper>
          rhythm guitar
          <SortingStyles>
            <Button
              disabled={false}
              icon={
                <Icon
                  color="#f6f6f6"
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
            <Button
              disabled={false}
              icon={
                <Icon
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
          </SortingStyles>
        </HeaderCellWrapper>
      </HeaderCellStyles>
    </HeaderRowStyles>
  </thead>
  <tbody>
    <BodyRowStyles
      key="0"
    >
      <BodyCellStyles
        key="0"
      >
        metallica
      </BodyCellStyles>
      <BodyCellStyles
        key="1"
      >
        james hatfield
      </BodyCellStyles>
      <BodyCellStyles
        key="2"
      >
        james hatfield
      </BodyCellStyles>
    </BodyRowStyles>
    <BodyRowStyles
      key="1"
    >
      <BodyCellStyles
        key="0"
      >
        slayer
      </BodyCellStyles>
      <BodyCellStyles
        key="1"
      >
        tom araya
      </BodyCellStyles>
      <BodyCellStyles
        key="2"
      >
        kerry king
      </BodyCellStyles>
    </BodyRowStyles>
    <BodyRowStyles
      key="2"
    >
      <BodyCellStyles
        key="0"
      >
        black sabbath
      </BodyCellStyles>
      <BodyCellStyles
        key="1"
      >
        ozzy osbourne
      </BodyCellStyles>
      <BodyCellStylesNone
        key="2"
      >
        none
      </BodyCellStylesNone>
    </BodyRowStyles>
    <BodyRowStyles
      key="3"
    >
      <BodyCellStyles
        key="0"
      >
        queen
      </BodyCellStyles>
      <BodyCellStyles
        key="1"
      >
        freddy mercury
      </BodyCellStyles>
      <BodyCellStylesNone
        key="2"
      >
        none
      </BodyCellStylesNone>
    </BodyRowStyles>
    <BodyRowStyles
      key="4"
    >
      <BodyCellStyles
        key="0"
      >
        led zeppelin
      </BodyCellStyles>
      <BodyCellStyles
        key="1"
      >
        robert plant
      </BodyCellStyles>
      <BodyCellStylesNone
        key="2"
      >
        none
      </BodyCellStylesNone>
    </BodyRowStyles>
    <BodyRowStyles
      key="5"
    >
      <BodyCellStyles
        key="0"
      >
        deep purple
      </BodyCellStyles>
      <BodyCellStyles
        key="1"
      >
        ian gillan
      </BodyCellStyles>
      <BodyCellStylesNone
        key="2"
      >
        none
      </BodyCellStylesNone>
    </BodyRowStyles>
  </tbody>
</TableStyles>
`;

exports[`renders table with width auto correctly: enzyme.shallow 1`] = `
<TableStyles
  data={
    Object {
      "body": Array [
        Object {
          "band": "metallica",
          "bass": "robert trujillo",
          "drums": "lars ulrich",
          "lead guitar": "kirk hammet",
          "rhythm guitar": "james hatfield",
          "singer": "james hatfield",
        },
        Object {
          "band": "slayer",
          "bass": "tom araya",
          "drums": "paul bostaph",
          "lead guitar": "gary holt",
          "rhythm guitar": "kerry king",
          "singer": "tom araya",
        },
        Object {
          "band": "black sabbath",
          "bass": "geezer butler",
          "drums": "bill ward",
          "lead guitar": "tommy iommi",
          "singer": "ozzy osbourne",
        },
        Object {
          "band": "queen",
          "bass": "john deacon",
          "drums": "roger taylor",
          "keyboard": "freddy mercury",
          "lead guitar": "brian may",
          "singer": "freddy mercury",
        },
        Object {
          "band": "led zeppelin",
          "bass": "john paul johns",
          "drums": "bonzo bonham",
          "keyboard": "john paul johns",
          "lead guitar": "jimmy page",
          "singer": "robert plant",
        },
        Object {
          "band": "deep purple",
          "bass": "roger glover",
          "drums": "ian paice",
          "keyboard": "jon lord",
          "lead guitar": "ritchie blackmore",
          "singer": "ian gillan",
        },
      ],
      "header": Array [
        "band",
        "singer",
        "rhythm guitar",
        "lead guitar",
        "bass",
        "drums",
        "keyboard",
      ],
    }
  }
  width="auto"
>
  <thead>
    <HeaderRowStyles>
      <HeaderCellStyles
        key="0"
      >
        <HeaderCellWrapper>
          band
          <SortingStyles>
            <Button
              disabled={false}
              icon={
                <Icon
                  color="#f6f6f6"
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
            <Button
              disabled={false}
              icon={
                <Icon
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
          </SortingStyles>
        </HeaderCellWrapper>
      </HeaderCellStyles>
      <HeaderCellStyles
        key="1"
      >
        <HeaderCellWrapper>
          singer
          <SortingStyles>
            <Button
              disabled={false}
              icon={
                <Icon
                  color="#f6f6f6"
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
            <Button
              disabled={false}
              icon={
                <Icon
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
          </SortingStyles>
        </HeaderCellWrapper>
      </HeaderCellStyles>
      <HeaderCellStyles
        key="2"
      >
        <HeaderCellWrapper>
          rhythm guitar
          <SortingStyles>
            <Button
              disabled={false}
              icon={
                <Icon
                  color="#f6f6f6"
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
            <Button
              disabled={false}
              icon={
                <Icon
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
          </SortingStyles>
        </HeaderCellWrapper>
      </HeaderCellStyles>
      <HeaderCellStyles
        key="3"
      >
        <HeaderCellWrapper>
          lead guitar
          <SortingStyles>
            <Button
              disabled={false}
              icon={
                <Icon
                  color="#f6f6f6"
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
            <Button
              disabled={false}
              icon={
                <Icon
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
          </SortingStyles>
        </HeaderCellWrapper>
      </HeaderCellStyles>
      <HeaderCellStyles
        key="4"
      >
        <HeaderCellWrapper>
          bass
          <SortingStyles>
            <Button
              disabled={false}
              icon={
                <Icon
                  color="#f6f6f6"
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
            <Button
              disabled={false}
              icon={
                <Icon
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
          </SortingStyles>
        </HeaderCellWrapper>
      </HeaderCellStyles>
      <HeaderCellStyles
        key="5"
      >
        <HeaderCellWrapper>
          drums
          <SortingStyles>
            <Button
              disabled={false}
              icon={
                <Icon
                  color="#f6f6f6"
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
            <Button
              disabled={false}
              icon={
                <Icon
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
          </SortingStyles>
        </HeaderCellWrapper>
      </HeaderCellStyles>
      <HeaderCellStyles
        key="6"
      >
        <HeaderCellWrapper>
          keyboard
          <SortingStyles>
            <Button
              disabled={false}
              icon={
                <Icon
                  color="#f6f6f6"
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
            <Button
              disabled={false}
              icon={
                <Icon
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
          </SortingStyles>
        </HeaderCellWrapper>
      </HeaderCellStyles>
    </HeaderRowStyles>
  </thead>
  <tbody>
    <BodyRowStyles
      key="0"
    >
      <BodyCellStyles
        key="0"
      >
        metallica
      </BodyCellStyles>
      <BodyCellStyles
        key="1"
      >
        james hatfield
      </BodyCellStyles>
      <BodyCellStyles
        key="2"
      >
        james hatfield
      </BodyCellStyles>
      <BodyCellStyles
        key="3"
      >
        kirk hammet
      </BodyCellStyles>
      <BodyCellStyles
        key="4"
      >
        robert trujillo
      </BodyCellStyles>
      <BodyCellStyles
        key="5"
      >
        lars ulrich
      </BodyCellStyles>
      <BodyCellStylesNone
        key="6"
      >
        none
      </BodyCellStylesNone>
    </BodyRowStyles>
    <BodyRowStyles
      key="1"
    >
      <BodyCellStyles
        key="0"
      >
        slayer
      </BodyCellStyles>
      <BodyCellStyles
        key="1"
      >
        tom araya
      </BodyCellStyles>
      <BodyCellStyles
        key="2"
      >
        kerry king
      </BodyCellStyles>
      <BodyCellStyles
        key="3"
      >
        gary holt
      </BodyCellStyles>
      <BodyCellStyles
        key="4"
      >
        tom araya
      </BodyCellStyles>
      <BodyCellStyles
        key="5"
      >
        paul bostaph
      </BodyCellStyles>
      <BodyCellStylesNone
        key="6"
      >
        none
      </BodyCellStylesNone>
    </BodyRowStyles>
    <BodyRowStyles
      key="2"
    >
      <BodyCellStyles
        key="0"
      >
        black sabbath
      </BodyCellStyles>
      <BodyCellStyles
        key="1"
      >
        ozzy osbourne
      </BodyCellStyles>
      <BodyCellStylesNone
        key="2"
      >
        none
      </BodyCellStylesNone>
      <BodyCellStyles
        key="3"
      >
        tommy iommi
      </BodyCellStyles>
      <BodyCellStyles
        key="4"
      >
        geezer butler
      </BodyCellStyles>
      <BodyCellStyles
        key="5"
      >
        bill ward
      </BodyCellStyles>
      <BodyCellStylesNone
        key="6"
      >
        none
      </BodyCellStylesNone>
    </BodyRowStyles>
    <BodyRowStyles
      key="3"
    >
      <BodyCellStyles
        key="0"
      >
        queen
      </BodyCellStyles>
      <BodyCellStyles
        key="1"
      >
        freddy mercury
      </BodyCellStyles>
      <BodyCellStylesNone
        key="2"
      >
        none
      </BodyCellStylesNone>
      <BodyCellStyles
        key="3"
      >
        brian may
      </BodyCellStyles>
      <BodyCellStyles
        key="4"
      >
        john deacon
      </BodyCellStyles>
      <BodyCellStyles
        key="5"
      >
        roger taylor
      </BodyCellStyles>
      <BodyCellStyles
        key="6"
      >
        freddy mercury
      </BodyCellStyles>
    </BodyRowStyles>
    <BodyRowStyles
      key="4"
    >
      <BodyCellStyles
        key="0"
      >
        led zeppelin
      </BodyCellStyles>
      <BodyCellStyles
        key="1"
      >
        robert plant
      </BodyCellStyles>
      <BodyCellStylesNone
        key="2"
      >
        none
      </BodyCellStylesNone>
      <BodyCellStyles
        key="3"
      >
        jimmy page
      </BodyCellStyles>
      <BodyCellStyles
        key="4"
      >
        john paul johns
      </BodyCellStyles>
      <BodyCellStyles
        key="5"
      >
        bonzo bonham
      </BodyCellStyles>
      <BodyCellStyles
        key="6"
      >
        john paul johns
      </BodyCellStyles>
    </BodyRowStyles>
    <BodyRowStyles
      key="5"
    >
      <BodyCellStyles
        key="0"
      >
        deep purple
      </BodyCellStyles>
      <BodyCellStyles
        key="1"
      >
        ian gillan
      </BodyCellStyles>
      <BodyCellStylesNone
        key="2"
      >
        none
      </BodyCellStylesNone>
      <BodyCellStyles
        key="3"
      >
        ritchie blackmore
      </BodyCellStyles>
      <BodyCellStyles
        key="4"
      >
        roger glover
      </BodyCellStyles>
      <BodyCellStyles
        key="5"
      >
        ian paice
      </BodyCellStyles>
      <BodyCellStyles
        key="6"
      >
        jon lord
      </BodyCellStyles>
    </BodyRowStyles>
  </tbody>
</TableStyles>
`;

exports[`renders table with width stretched correctly: enzyme.shallow 1`] = `
<TableStyles
  data={
    Object {
      "body": Array [
        Object {
          "band": "metallica",
          "bass": "robert trujillo",
          "drums": "lars ulrich",
          "lead guitar": "kirk hammet",
          "rhythm guitar": "james hatfield",
          "singer": "james hatfield",
        },
        Object {
          "band": "slayer",
          "bass": "tom araya",
          "drums": "paul bostaph",
          "lead guitar": "gary holt",
          "rhythm guitar": "kerry king",
          "singer": "tom araya",
        },
        Object {
          "band": "black sabbath",
          "bass": "geezer butler",
          "drums": "bill ward",
          "lead guitar": "tommy iommi",
          "singer": "ozzy osbourne",
        },
        Object {
          "band": "queen",
          "bass": "john deacon",
          "drums": "roger taylor",
          "keyboard": "freddy mercury",
          "lead guitar": "brian may",
          "singer": "freddy mercury",
        },
        Object {
          "band": "led zeppelin",
          "bass": "john paul johns",
          "drums": "bonzo bonham",
          "keyboard": "john paul johns",
          "lead guitar": "jimmy page",
          "singer": "robert plant",
        },
        Object {
          "band": "deep purple",
          "bass": "roger glover",
          "drums": "ian paice",
          "keyboard": "jon lord",
          "lead guitar": "ritchie blackmore",
          "singer": "ian gillan",
        },
      ],
      "header": Array [
        "band",
        "singer",
        "rhythm guitar",
        "lead guitar",
        "bass",
        "drums",
        "keyboard",
      ],
    }
  }
  width="stretched"
>
  <thead>
    <HeaderRowStyles>
      <HeaderCellStyles
        key="0"
      >
        <HeaderCellWrapper>
          band
          <SortingStyles>
            <Button
              disabled={false}
              icon={
                <Icon
                  color="#f6f6f6"
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
            <Button
              disabled={false}
              icon={
                <Icon
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
          </SortingStyles>
        </HeaderCellWrapper>
      </HeaderCellStyles>
      <HeaderCellStyles
        key="1"
      >
        <HeaderCellWrapper>
          singer
          <SortingStyles>
            <Button
              disabled={false}
              icon={
                <Icon
                  color="#f6f6f6"
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
            <Button
              disabled={false}
              icon={
                <Icon
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
          </SortingStyles>
        </HeaderCellWrapper>
      </HeaderCellStyles>
      <HeaderCellStyles
        key="2"
      >
        <HeaderCellWrapper>
          rhythm guitar
          <SortingStyles>
            <Button
              disabled={false}
              icon={
                <Icon
                  color="#f6f6f6"
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
            <Button
              disabled={false}
              icon={
                <Icon
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
          </SortingStyles>
        </HeaderCellWrapper>
      </HeaderCellStyles>
      <HeaderCellStyles
        key="3"
      >
        <HeaderCellWrapper>
          lead guitar
          <SortingStyles>
            <Button
              disabled={false}
              icon={
                <Icon
                  color="#f6f6f6"
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
            <Button
              disabled={false}
              icon={
                <Icon
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
          </SortingStyles>
        </HeaderCellWrapper>
      </HeaderCellStyles>
      <HeaderCellStyles
        key="4"
      >
        <HeaderCellWrapper>
          bass
          <SortingStyles>
            <Button
              disabled={false}
              icon={
                <Icon
                  color="#f6f6f6"
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
            <Button
              disabled={false}
              icon={
                <Icon
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
          </SortingStyles>
        </HeaderCellWrapper>
      </HeaderCellStyles>
      <HeaderCellStyles
        key="5"
      >
        <HeaderCellWrapper>
          drums
          <SortingStyles>
            <Button
              disabled={false}
              icon={
                <Icon
                  color="#f6f6f6"
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
            <Button
              disabled={false}
              icon={
                <Icon
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
          </SortingStyles>
        </HeaderCellWrapper>
      </HeaderCellStyles>
      <HeaderCellStyles
        key="6"
      >
        <HeaderCellWrapper>
          keyboard
          <SortingStyles>
            <Button
              disabled={false}
              icon={
                <Icon
                  color="#f6f6f6"
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
            <Button
              disabled={false}
              icon={
                <Icon
                  icon="ChevronSvg"
                  size="xs"
                />
              }
              iconDirection="left"
              mode="flat"
              onClick={[Function]}
              size="s"
              stretched={false}
              type="secondary"
            />
          </SortingStyles>
        </HeaderCellWrapper>
      </HeaderCellStyles>
    </HeaderRowStyles>
  </thead>
  <tbody>
    <BodyRowStyles
      key="0"
    >
      <BodyCellStyles
        key="0"
      >
        metallica
      </BodyCellStyles>
      <BodyCellStyles
        key="1"
      >
        james hatfield
      </BodyCellStyles>
      <BodyCellStyles
        key="2"
      >
        james hatfield
      </BodyCellStyles>
      <BodyCellStyles
        key="3"
      >
        kirk hammet
      </BodyCellStyles>
      <BodyCellStyles
        key="4"
      >
        robert trujillo
      </BodyCellStyles>
      <BodyCellStyles
        key="5"
      >
        lars ulrich
      </BodyCellStyles>
      <BodyCellStylesNone
        key="6"
      >
        none
      </BodyCellStylesNone>
    </BodyRowStyles>
    <BodyRowStyles
      key="1"
    >
      <BodyCellStyles
        key="0"
      >
        slayer
      </BodyCellStyles>
      <BodyCellStyles
        key="1"
      >
        tom araya
      </BodyCellStyles>
      <BodyCellStyles
        key="2"
      >
        kerry king
      </BodyCellStyles>
      <BodyCellStyles
        key="3"
      >
        gary holt
      </BodyCellStyles>
      <BodyCellStyles
        key="4"
      >
        tom araya
      </BodyCellStyles>
      <BodyCellStyles
        key="5"
      >
        paul bostaph
      </BodyCellStyles>
      <BodyCellStylesNone
        key="6"
      >
        none
      </BodyCellStylesNone>
    </BodyRowStyles>
    <BodyRowStyles
      key="2"
    >
      <BodyCellStyles
        key="0"
      >
        black sabbath
      </BodyCellStyles>
      <BodyCellStyles
        key="1"
      >
        ozzy osbourne
      </BodyCellStyles>
      <BodyCellStylesNone
        key="2"
      >
        none
      </BodyCellStylesNone>
      <BodyCellStyles
        key="3"
      >
        tommy iommi
      </BodyCellStyles>
      <BodyCellStyles
        key="4"
      >
        geezer butler
      </BodyCellStyles>
      <BodyCellStyles
        key="5"
      >
        bill ward
      </BodyCellStyles>
      <BodyCellStylesNone
        key="6"
      >
        none
      </BodyCellStylesNone>
    </BodyRowStyles>
    <BodyRowStyles
      key="3"
    >
      <BodyCellStyles
        key="0"
      >
        queen
      </BodyCellStyles>
      <BodyCellStyles
        key="1"
      >
        freddy mercury
      </BodyCellStyles>
      <BodyCellStylesNone
        key="2"
      >
        none
      </BodyCellStylesNone>
      <BodyCellStyles
        key="3"
      >
        brian may
      </BodyCellStyles>
      <BodyCellStyles
        key="4"
      >
        john deacon
      </BodyCellStyles>
      <BodyCellStyles
        key="5"
      >
        roger taylor
      </BodyCellStyles>
      <BodyCellStyles
        key="6"
      >
        freddy mercury
      </BodyCellStyles>
    </BodyRowStyles>
    <BodyRowStyles
      key="4"
    >
      <BodyCellStyles
        key="0"
      >
        led zeppelin
      </BodyCellStyles>
      <BodyCellStyles
        key="1"
      >
        robert plant
      </BodyCellStyles>
      <BodyCellStylesNone
        key="2"
      >
        none
      </BodyCellStylesNone>
      <BodyCellStyles
        key="3"
      >
        jimmy page
      </BodyCellStyles>
      <BodyCellStyles
        key="4"
      >
        john paul johns
      </BodyCellStyles>
      <BodyCellStyles
        key="5"
      >
        bonzo bonham
      </BodyCellStyles>
      <BodyCellStyles
        key="6"
      >
        john paul johns
      </BodyCellStyles>
    </BodyRowStyles>
    <BodyRowStyles
      key="5"
    >
      <BodyCellStyles
        key="0"
      >
        deep purple
      </BodyCellStyles>
      <BodyCellStyles
        key="1"
      >
        ian gillan
      </BodyCellStyles>
      <BodyCellStylesNone
        key="2"
      >
        none
      </BodyCellStylesNone>
      <BodyCellStyles
        key="3"
      >
        ritchie blackmore
      </BodyCellStyles>
      <BodyCellStyles
        key="4"
      >
        roger glover
      </BodyCellStyles>
      <BodyCellStyles
        key="5"
      >
        ian paice
      </BodyCellStyles>
      <BodyCellStyles
        key="6"
      >
        jon lord
      </BodyCellStyles>
    </BodyRowStyles>
  </tbody>
</TableStyles>
`;
